console/module/data/src/main/java/org/eclipse/kapua/app/console/module/data/client/GridSelectionChangedListener.java

```diff
@@ -0,0 +1,47 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Eurotech and/or its affiliates and others
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Eurotech - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.kapua.app.console.module.data.client;
+
+import com.extjs.gxt.ui.client.data.ModelData;
+import com.extjs.gxt.ui.client.event.Listener;
+import com.extjs.gxt.ui.client.event.SelectionChangedEvent;
+import com.extjs.gxt.ui.client.widget.grid.GridSelectionModel;
+
+public class GridSelectionChangedListener<M extends ModelData> implements Listener<SelectionChangedEvent<M>> {
+    private GridSelectionModel<M> selectionModel;
+    protected boolean selectedAgain;
+    private M selectedItem;
+
+    public GridSelectionChangedListener() {
+        super();
+    }
+
+    public void setSelectionModel(GridSelectionModel<M> selectionModel) {
+        this.selectionModel = selectionModel;
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public void handleEvent(SelectionChangedEvent<M> be) {
+        if (selectionModel.getSelectedItem() != null && selectedAgain == false) {
+            selectedItem = selectionModel.getSelectedItem();
+            selectionModel.select(false, selectedItem);
```

Comment:
Can you restructure this if stament as selectionModel.select(false, selectedItem); is repeated in all sentences.